facebook/ThreatExchange
# Bucket hasher

Summary
---------

This PR enables a user to connect HMA to a local bank such that all images uploaded to that bank will be processed by HMA without having to call a separate API.

Future work:
1. Enable user to filter to specific folders to upload

Test Plan
---------

1. Create a new bucket `jesses-separate-aws-bucket` in the UI that is not controlled by terraform to simulate a separate bucket owned by a partner
2. Add the following line to my `terraform.tfvars` file:
```
local_image_buckets = [{"name": "jesses-separate-aws-bucket", "arn" : "arn:aws:s3:::jesses-separate-aws-bucket"}]
```
3. `terraform -chdir=terraform apply`
4. using the AWS UI upload images to the bucket
5. Through logs watch them be processed by the local_bucket_hasher lambda, added to the sqs queue, and then loaded and hashed by the pdq_hasher: <img width="1352" alt="Screen Shot 2021-07-20 at 7 11 09 PM" src="https://user-images.githubusercontent.com/24800630/126406600-55a5e231-1d00-4e52-a543-2ef0f88ae343.png">
6. Uploaded images using the API to ensure that path still works
7. Uploaded images to HMA s3 bucket directly through UI to ensure that path still works
